### ISSUE TITLE
Unstable container images using stable tag pattern 

### ISSUE THREAD

**Issue by dghubble** [tailscale/tailscale#6613]
### What is the issue?

According to Tailsclale docs on [Dockerhub](https://hub.docker.com/r/tailscale/tailscale):

* get a specific stable version (e.g. v1.20.1, v1.20)
* get a specific unstable version (e.g. unstable-v1.33.159, unstable-v1.33)

At the time of this writing, the latest stable release is v1.32.3, but on Dockerhub or GHCR there are images `v1.33.158` (as well as `unstable-v1.33.159`). I noticed this because I use bots (like renovate) to manage image versions across my infra, and tailscale stands out - it proposes upgrading from `v1.32.3` to `v1.33.158`.

That raises some interesting questions. If `v1.33` is still under development, is `v1.33.158` stable or unstable? I believe its your unstable / development work. What happens when `v1.33` is officially stabalized? You're going to release a `v1.33` and `v1.33.0`. Great. And then stable patch versions become `v1.33.1`, etc. But now this tagging pattern overlaps with your unstable versions.... as early as `v1.33.19`.

Basically, this tag pattern seems wrong and its also causing bots to suggest updating to your unstable/development tags.

### Steps to reproduce

Search with tag filters on [Dockerhub](https://hub.docker.com/r/tailscale/tailscale/tags) or https://github.com/tailscale/tailscale/pkgs/container/tailscale

Find `v1.33.19` - https://hub.docker.com/r/tailscale/tailscale/tags?page=1&name=v1.33.19 - this is going to collide with a future "stable" v1.33.19 release.

### Are there any recent changes that introduced the issue?

I thought this may have started with v1.33 development. The tags within v1.32.x just include v1.32.0, v1.32.1, v1.32.2, v1.32.3. Nice. But it may have been occuring back in v1.31 as well: v1.31.6, v1.31.7, ... v1.31.101(?)

**Comment by DentonGentry** [comment]
The `unstable-*` dockerhub tagging convention was adopted November 22, 2022: https://tailscale.com/changelog/#2022-11-22-service

Earlier builds tagged before that point remain on dockerhub. We're expecting to release 1.34.0 tomorrow, and the 1.35.x builds will all be tagged `unstable-v1.35.x`

If the point of parsing out release numbers is to use the latest stable image, `tailscale/tailscale:stable` is available and kept updated.

**Comment by dghubble** [comment]
@DentonGentry awesome, that helps me understand what's going on.

Sounds like this will be resolved with v1.34.0 and onward! I appreciate the `stable` tag exists too.
